prow/pod-utils/downwardapi/jobspec.go

```diff
@@ -122,18 +122,22 @@ func EnvForSpec(spec JobSpec) (map[string]string, error) {
 		return env, nil
 	}
 
-	env[repoOwnerEnv] = spec.Refs.Org
-	env[repoNameEnv] = spec.Refs.Repo
-	env[pullBaseRefEnv] = spec.Refs.BaseRef
-	env[pullBaseShaEnv] = spec.Refs.BaseSHA
-	env[pullRefsEnv] = spec.Refs.String()
+	if spec.Refs != nil {
```

Comment:
This means a caller is doing it wrong. No job other than periodics should have `nil` refs. 

/close